noir-projects/noir-contracts/contracts/token_contract/src/types/balances_map.nr

```diff
@@ -70,8 +71,7 @@ impl<T> BalancesMap<T, &mut PrivateContext> {
             let context = self.map.context;
 
             // We fetch the nullifier public key hash from the registry / from our PXE
-            let owner_npk_m_hash = get_current_public_keys(context, owner).npk_m.hash();
-            let mut addend_note = T::new(addend, owner_npk_m_hash);
+            let mut addend_note = T::new(addend, owner_npk_m.hash());
```

Comment:
This makes me think, do we ever use the `npk`, or is it always its hash?

Reply:
I checked all the occurrences and we always just compute a hash from npk_m. This make sense because the nullifier keys are only used when dealing with notes and there we just store npk_m_hash upon note creation and then when nullifying we just call context.request_nsk_app(npk_m_hash).

It would really make sense to make get_current_public_keys return only the npk_m_hash. It would make naming a bit tricky though.